python/whylogs/api/whylabs/session/session_manager.py

```diff
@@ -0,0 +1,285 @@
+            self.org_id = config.get_or_prompt_org_id(persist=True)
+        else:
+            logger.warning("No org id found in session or configuration, will not be able to send data to whylabs.")
+
+        if is_notebook():
+            log_if_notebook(f"Initialized whylabs session with for org {self.org_id}")
+
+    def get_type(self) -> SessionType:
+        return SessionType.WHYLABS
+
+    def upload_reference_profiles(
+        self, profile_aliases: Dict[str, ResultSet]
+    ) -> Union[UploadReferenceResult, NotSupported]:
+        # TODO support soon
+        return NotSupported()
+
+
+class SessionManager:
+    __instance: Optional["SessionManager"] = None
+
+    def __init__(self, type: SessionType = SessionType.LOCAL):
+        self._config = SessionConfig()
+        client_config = Configuration()
+        client_config.host = self._config.get_whylabs_endpoint()
+        self._whylabs_client = ApiClient(client_config)
+
+        self.session: Optional[Session] = None
+        if type == SessionType.LOCAL:
+            pass
+        elif type == SessionType.WHYLABS_ANONYMOUS:
+            self.session = GuestSession(self._config, self._whylabs_client)
+        elif type == SessionType.WHYLABS:
+            self.session = ApiKeySession(self._config)
+
+    @staticmethod
+    def init(type: SessionType = SessionType.LOCAL) -> None:
+        if SessionManager.__instance is None:
+            init_notebook_logging()
+            SessionManager.__instance = SessionManager(type=type)
+        else:
+            logger.warning("SessionManager is already initialized. Ignoring call to init()")
+
+    @staticmethod
+    def reset() -> None:
+        SessionManager.__instance = None
+
+    @staticmethod
+    def get_instance() -> Optional["SessionManager"]:
+        return SessionManager.__instance
+
+    @staticmethod
+    def is_active() -> bool:
+        return SessionManager.get_instance() is not None
+
+
+def init(session_type: Union[SessionType, str] = SessionType.LOCAL, reinit: bool = False) -> None:
+    if reinit:
+        SessionManager.reset()
+
+    if isinstance(session_type, str):
+        try:
+            _session_type = SessionType[session_type.upper()]
+        except KeyError:
+            logger.error(f"Invalid session type {session_type}. Valid values are {SessionType.__members__.keys()}")
+            return
+    else:
+        _session_type = session_type
+
+    try:
+        SessionManager.init(type=_session_type)
+    except PermissionError as e:
+        logger.warning("Could not create or read configuration file for session. Profiles won't be uploaded.", e)
+    except Exception as e:
+        logger.warning("Could not initialize session", e)
+
+
+def get_current_session() -> Optional[Session]:
+    manager = SessionManager.get_instance()
+    if manager is not None:
+        return manager.session
+
+    logger.warning("No session is initialized. Call whylogs.init(anonymous=True) to create a guest session.")
```

Comment:
The code for production logging isn't here yet, but if you were to do it with something like :

```
why.init(api_ke='...', ...)
```

Then you would end up seeing this message (which is currently tailored towards guest sessions since the production init doesn't exist yet). I think the assumption there is that you ran init with the intent of having it actually work. The message has to change after I add the ability to actually do api key init of course.

You would see this though even if you didn't use a session, which I agree isn't right. I updated it to use the `log_if_notebook` util so that it's only shown in a notebook environment, and I added a flag so that it only happens once.